### ISSUE TITLE
Unable to use output with space in identifier

### ISSUE THREAD

**Issue by dhaines** [hashicorp/terraform#16843]
### Terraform Version
Terraform v0.11.1
+ provider.openstack v1.0.0

### Terraform Configuration Files
From third-party code:

```
output "router id" {
  value = "${openstack_networking_router_v2.bosh_router.id}"
}
```

From my own code (multiple examples, not comprehensive):

```
output "net_id" {
  value = "${module.bosh-init-tf.network id}"
}
```
```
output "net_id" {
  value = "${module.bosh-init-tf."network id"}"
}
```
```
output "net_id" {
  value = "${module.bosh-init-tf.network\ id}"
}
```

etc.

### Output
Variously:

> Error: Failed to load root config module: Error loading /home/hainesd/devops/trebuchet/terraform/bosh/jumpbox.tf: Error reading config for output net_id: parse error at 1:31: expected "}" but found "id"

> Error: Failed to load root config module: Error parsing /home/hainesd/devops/trebuchet/terraform/bosh/jumpbox.tf: At 31:42: illegal char escape

> Error: Failed to load root config module: Error loading /home/hainesd/devops/trebuchet/terraform/bosh/jumpbox.tf: Error reading config for output net_id: parse error at 1:30: expected "}" but found invalid sequence "\\"

etc.

### Expected Behavior
One of the syntaxes I've tried would work to interpolate the third-party output or the third-party output would itself throw an error.

### Actual Behavior
Errors generated for everything I've tried, nothing relating to the output definition.

### Steps to Reproduce
```
terraform init
terraform apply
```

### Perceived erroneous behavior
It seems to me that there should be some way to escape a space in an output identifier or that spaces should be illegal characters in output identifiers; having neither be true makes the language syntactically incomplete and/or incoherent in some situations.

**Comment by apparentlymart** [comment]
Hi @dhaines! Sorry for this strange behavior.

It is indeed supposed to be forbidden to use a space in an output. Looks like there's some missing validation here.

**Comment by apparentlymart** [comment]
Hi again @dhaines! Sorry for the long silence.

I've just confirmed that the `output` block name is now validated correctly in v0.12.0-alpha1, and that fix will be included in the forthcoming v0.12.0 release.

```
Error: Invalid output name

  on output-name-validation.tf line 1, in output "router id":
   1: output "router id" {

A name must start with a letter and may contain only letters, digits,
underscores, and dashes.
```

This means that the third-party configuration you mentioned will produce an error after upgrading to Terraform 0.12. If you are able to reach the developers of that module, I'd suggest letting them know about this so that they might address it. As you saw, there was previously no way to actually use an output whose name contained a space so hopefully the developer of that module can rename the output to `router_id` instead.

Thanks for reporting this! Since it's now fixed in master, I'm going to close out this issue.
